lib/modules/manager/carthage/extract.ts

```diff
@@ -0,0 +1,155 @@
+import { logger } from '../../../logger';
+import { getSiblingFileName, localPathExists } from '../../../util/fs';
+import { newlineRegex, regEx } from '../../../util/regex';
+import { coerceString } from '../../../util/string';
+import { CarthageDatasource } from '../../datasource/carthage';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { GitlabTagsDatasource } from '../../datasource/gitlab-tags';
+import type { PackageDependency, PackageFileContent } from '../types';
+import type { ManagerData, ParsedLine } from './types';
+
+const regexMappings = [
+  regEx(
+    `^\\s*(?<type>[^\\s]+)\\s+(['"])(?<url>[^'"]+)(['"])\\s*(['"])(?<version>[^'"]+)(['"])\\s*$`
+  ),
+  regEx(
+    `^\\s*(?<type>[^\\s]+)\\s+(['"])(?<url>[^'"]+)(['"])\\s*(?<condition>>=|==|~>)\\s*(?<version>[^\\s]+)\\s*$`
+  ),
```

Comment:
You shouldn't need to decide this here, but only return `currentValue` with `== 1.0.0` or `>= 1.0.0`. The analysis of the `currentValue` is the job of the versioning

This allows you to simplify to one regex